### PR TITLE
DNM Promote ec2_vpc_peer and ec2_vpc_peering_info modules

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

DNM Promote ec2_vpc_peer and ec2_vpc_peering_info modules

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/ec2_vpc_peer.py [hunk 1]
```diff
@@ -0,0 +1,613 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+module: ec2_vpc_peer
+short_description: create, delete, accept, and reject VPC peering connections between two VPCs.
+version_added: 1.0.0
```
Comment (GomathiselviS) [inline plugins/modules/ec2_vpc_peer.py:10]: version_added_collection: community.aws

File: plugins/modules/ec2_vpc_peering_info.py [hunk 2]
```diff
@@ -0,0 +1,387 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+module: ec2_vpc_peering_info
+short_description: Retrieves AWS VPC Peering details using AWS methods
+version_added: 1.0.0
```
Comment (GomathiselviS) [inline plugins/modules/ec2_vpc_peering_info.py:10]: version_added_collection: community.aws

File: tests/integration/targets/ec2_vpc_peer/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,657 @@
...
+    aws_caller_info:
+    register: aws_caller_info
+
+  - name: Store Account ID for later use
+    ansible.builtin.set_fact:
+      account_id: '{{ aws_caller_info.account }}'
+
+  # ============================================================
+  - name: Fetch EC2 VPC Peering Connections in check_mode
+    community.aws.ec2_vpc_peering_info:
```
Comment (GomathiselviS) [inline tests/integration/targets/ec2_vpc_peer/tasks/main.yml:19]:
```suggestion
    amazon.aws.ec2_vpc_peering_info:
```

Comment (GomathiselviS) [reply tests/integration/targets/ec2_vpc_peer/tasks/main.yml:19]: The collection name has to be changed for the following tasks as well

**Review by GomathiselviS — APPROVED** [review]

**Review by mandar242 — APPROVED** [review]

**Pull request merged**